src/actions/hid.action.ts

```diff
@@ -290,13 +291,32 @@ export const hidActionsThunk = {
         product_id: keyboard.getInformation().productId,
         product_name: keyboard.getInformation().productName,
       });
-      dispatch(
-        HidActions.updateBleMicroPro(
-          keyboard
-            .getInformation()
-            .productName.includes(PRODUCT_PREFIX_FOR_BLE_MICRO_PRO)
-        )
-      );
+
+      const isBleMicroPro = keyboard
+        .getInformation()
+        .productName.includes(PRODUCT_PREFIX_FOR_BLE_MICRO_PRO);
+
+      // Override custom keycode list if the keyboard uses BLE Micro Pro
+      const customKeycodes = !isBleMicroPro
+        ? entities.keyboardDefinition?.customKeycodes
+        : bmpKeyInfoList.map((k) => {
+            return {
+              name: k.keycodeInfo.label,
+              title: k.desc,
+              shortName: k.keycodeInfo.name.short,
+            };
+          });
+
+      if (isBleMicroPro) {
+        dispatch(HidActions.updateBleMicroPro(isBleMicroPro));
+        dispatch(
+          StorageActions.updateKeyboardDefinition({
+            ...entities.keyboardDefinition,
+            customKeycodes: customKeycodes,
+          })
+        );
+      }
```

Comment:
```suggestion
      dispatch(HidActions.updateBleMicroPro(isBleMicroPro));

      // Override custom keycode list if the keyboard uses BLE Micro Pro
      const customKeycodes = !isBleMicroPro
        ? entities.keyboardDefinition?.customKeycodes
        : bmpKeyInfoList.map((k) => {
            return {
              name: k.keycodeInfo.label,
              title: k.desc,
              shortName: k.keycodeInfo.name.short,
            };
          });

      if (isBleMicroPro) {
        dispatch(
          StorageActions.updateKeyboardDefinition({
            ...entities.keyboardDefinition,
            customKeycodes: customKeycodes,
          })
        );
      }
```

@sekigon-gonnoc In the diff you have created, the HidActions.updateBleMicroPro() function is called only when the isBleMicroPro is true. If a user connects a keyboard with BMP, the keyboard is disconnected and then other keyboard using other MCU (not BMP) is connected, the bleMicroPro flag is still true. That is, the BMP feature will be shown on Remap for the keyboard which BMP is not used, I guess. Could you modify the code so that the `updateBleMicroPro` function is called regardless of the `isBleMicroPro` value?